pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -144,8 +180,37 @@ func (s *Service) Disconnected(_ network.Network, c network.Conn) {
 	delete(s.receivedHandshakes, c.RemotePeer())
 }
 
+func (s *Service) checkSyn(syn *pb.Syn) error {
+	if syn.NetworkID != s.networkID {
+		return ErrNetworkIDIncompatible
+	}
+
+	recoveredPK, err := crypto.Recover(syn.BzzAddress.Signature, append(syn.BzzAddress.Underlay, strconv.FormatUint(syn.NetworkID, 10)...))
+	if err != nil {
+		return ErrInvalidSignature
+	}
+
+	recoveredOverlay := crypto.NewOverlayAddress(*recoveredPK, syn.NetworkID)
```

Comment:
I checked this package and found that it is not conforming to spec, as the overlay is the hash of the eth address + network id not public key +. network iD